server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1309,20 +1307,22 @@ public CacheHelper getReaderCacheHelper() {
     }
 
     public void close(String reason, boolean flushEngine) throws IOException {
-        synchronized (mutex) {
-            try {
-                changeState(IndexShardState.CLOSED, reason);
-            } finally {
-                final Engine engine = this.currentEngineReference.getAndSet(null);
+        synchronized (engineMutex) {
```

Comment:
I think this means that we now risk waiting for an engine warmer during close? I.e., partially undo the effect of #45263?